src/tslint.ts

```diff
@@ -63,24 +63,32 @@ module Lint {
             var failures: RuleFailure[] = [];
             var sourceFile = Lint.getSourceFile(this.fileName, this.source);
 
-            // walk the code first to find all the intervals where rules are disabled
-            var rulesWalker = new EnableDisableRulesWalker(sourceFile, {ruleName: "", disabledIntervals: []});
-            rulesWalker.walk(sourceFile);
-            var enableDisableRuleMap = rulesWalker.enableDisableRuleMap;
-
-            var rulesDirectory = this.getRelativePath(this.options.rulesDirectory);
-            var configuration = this.options.configuration.rules;
-            var configuredRules = Lint.loadRules(configuration, enableDisableRuleMap, rulesDirectory);
-            for (var i = 0; i < configuredRules.length; ++i) {
-                var rule = configuredRules[i];
-                if (rule.isEnabled()) {
-                    var ruleFailures = rule.apply(sourceFile);
-                    ruleFailures.forEach((ruleFailure) => {
-                        if (!this.containsRule(failures, ruleFailure)) {
-                            failures.push(ruleFailure);
-                        }
-                    });
+            try {
+                // walk the code first to find all the intervals where rules are disabled
+                var rulesWalker = new EnableDisableRulesWalker(sourceFile, {ruleName: "", disabledIntervals: []});
+                rulesWalker.walk(sourceFile);
+                var enableDisableRuleMap = rulesWalker.enableDisableRuleMap;
+
+                var rulesDirectory = this.getRelativePath(this.options.rulesDirectory);
+                var configuration = this.options.configuration.rules;
+                var configuredRules = Lint.loadRules(configuration, enableDisableRuleMap, rulesDirectory);
+                for (var i = 0; i < configuredRules.length; ++i) {
+                    var rule = configuredRules[i];
+                    if (rule.isEnabled()) {
+                        var ruleFailures = rule.apply(sourceFile);
+                        ruleFailures.forEach((ruleFailure) => {
+                            if (!this.containsRule(failures, ruleFailure)) {
+                                failures.push(ruleFailure);
+                            }
+                        });
+                    }
                 }
+            } catch (e) {
+                return {
+                    failureCount: 1,
+                    format: "error",
+                    output: "This file did not successfully compile; please fix compilation errors and re-run tslint"
```

Comment:
this is incorrect. this should not be a string because this will throw an exception for people externally parsing the output via the JSON formatter or some other custom format. instead i think you should just create a single RuleFailure with an empty rule name and at position 0 with the error message "unable to compile file" and pass that into the formatter.